cmd/czds-dl/download.go

```diff
@@ -103,22 +110,32 @@ func main() {
 func addLinks(downloads []string) {
 	for _, dl := range downloads {
 		work.Add(1)
-		inputChan <- dl
-	}
-	close(inputChan)
```

Comment:
Closing `inputChan` helps prevent deadlock, was this removal intentional?

Reply:
It was intentional.  The channel is re-used later to requeue failed downloads.  Careful to add to the worker group so the task items doesn't come desynced.

Reply:
Unfortunately; I think that this introduced a deadlock possibility.

For example, say `client.GetLinks()` called from the `addLinks()` goroutine returns 1000 zones, and starts inserting them into `inputChan` which has a buffer of 100. Since there are more zones than fit in the buffer `addLinks()` will block when inserting into `inputChan` after it is full.
Now while `addLinks()` is blocked, if CZDS has some errors and zones need to be retried, all the threads will also block when attempting to insert the zones back into `inputChan` because `inputChan` is still at capacity, resulting in deadlock.

I'm trying to think of an elegant way to fix this. One possibility is to run the line `inputChan <- zi // requeue` in its own goroutine to not block the worker thread, but this may have the unintended consequence of starting a separate threads for every zone if CZDS is having a bad day.

Reply:
I think I understand your point.  You are saying that if and only if addLinks still has entries to add, but also all workers have failed on their fetch and try to re-add to the queue would you experience a deadlock.  Let me give this some thought as well.